### DISCUSSION TITLE
Actions org appears close to unmaintained

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#49316] [Actions]
@Akash1134 declined to reach out privately and instead asked me to use this forum.

I really didn't want to throw anyone under a 🚍, but so be it. 

This isn't meant to single out any particular repository in the https://github.com/actions/ organization, but I had to start somewhere, so I surveyed this discussion category. By far most recent complaints related to a single action:

## [actions/setup-java](https://github.com/actions/setup-java)

Here's a short list of issues (that have been poorly triaged/handled) with that repository:
https://github.com/actions/setup-java/issues/415
https://github.com/actions/setup-java/issues/408
https://github.com/actions/setup-java/issues/396
https://github.com/actions/setup-java/issues/389
https://github.com/actions/setup-java/issues/326
https://github.com/actions/setup-java/issues/286
https://github.com/actions/setup-java/issues/279#issuecomment-1013728659
https://github.com/actions/setup-java/issues/84
https://github.com/actions/setup-java/issues/72
https://github.com/actions/setup-java/issues/71

(Composed on a mobile phone using GitHub Mobile. – fwiw, the GitHub Mobile team is very responsive to feedback.)

Quick notes:
1. In this list is a request to support actions/cache. Afaict, that was done, but the ticket wasn't closed. 
2. Then there are a bunch of complaints that the caching doesn't work well. 
  - This point is what resulted in most of those complaints in this discussion forum. 
3. There was a bit of a mess involving the adopt openjdk rebrands, now, some of that is definitely their fault, but the actions organization "maintains" the repository and took the half measure for temurin while not handling its sibling. 
4. There are a couple of other requests to add additional distributions, and they haven't been handled in a timely manner. I'm not saying that they have to be accepted, a "no" is also a reasonable response. Purgatory isn't. 

---

I've been trying to speak privately to GitHub's staff for over a year and have notes about a whole bunch of these repositories. I've also sent PRs to many of them. I do try to contribute. In a number of repositories, my PRs are just sitting there ignored (along with PRs from others, I'm not special).

**Comment by rmunn** [comment]
And as of June 2025, they've updated the README in lots of actions to say "Thank you for your interest in this GitHub action, however, right now we are not taking contributions." Their explanation:

> We continue to focus our resources on strategic areas that help our customers be successful while making developers' lives easier. While GitHub Actions remains a key part of this vision, we are allocating resources towards other areas of Actions and are not taking contributions to this repository at this time. The GitHub public roadmap is the best place to follow along for any updates on features we’re working on and what stage they’re in.

Translated from corporate-speak, that means "GitHub Actions isn't making us any money so we're not going to work on it any more". Here's the rest of what they say, with my comments interspersed:

> We are taking the following steps to better direct requests related to GitHub Actions, including:
> 
> 1. We will be directing questions and support requests to our [Community Discussions area](https://github.com/orgs/community/discussions/categories/actions)

Where they will be ignored by MS staff, and only answered by the community.

> 2. High Priority bugs can be reported through Community Discussions or you can report these to our support team https://support.github.com/contact/bug-report.

Where they will, based on past experience, also be ignored.

> 3. Security Issues should be handled as per our [security.md](https://github.com/actions/checkout/blob/main/security.md)

These will be fixed, because MS kind of has to.

> We will still provide security updates for this project and fix major breaking changes during this time.

There you go: we will fix "major breaking changes". Annoyances like https://github.com/actions/upload-artifact/issues/426 will continue to be ignored.

> You are welcome to still raise bugs in this repo.

Where they will be ignored.

Sorry, MS staff who may happen to see this. It's not you I'm annoyed with, it's your management. Who won't see this message because they don't care what GitHub users think.
